src/System/NowPlaying.m

```diff
@@ -113,22 +117,27 @@ - (void)updateApp
             NSString *appBundleIdentifier = nil;
             NSString *appName = nil;
             NSImage *appIcon = nil;
+            
+            NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
```

Comment:
This is probably the wrong place to be storing User Defaults. I would move them to the NowPlayingWidget itself.

Reply:
It tries to load the appBundleIdentifier of the last music player from the NSUserDefaults only if MRMediaRemoteGetNowPlayingClient returns nil.

It also saves the last music player used in NSUserDefaults on each updateApp.

It keeps the current implementation logic filling the appBundleIdentifier to get the name+icon of the application.

Let me know if you prefer to duplicate this logic in the NowPlayingWidget also.